src/Microsoft.XmlSerializer.Generator/src/Sgen.cs

```diff
@@ -253,9 +253,18 @@ private void GenerateFile(List<string> typeNames, string assemblyName, bool prox
                         File.Delete(codePath);
                     }
 
-                    using (FileStream fs = File.Create(codePath))
+                    using (MemoryStream ms = new MemoryStream())
```

Comment:
I don't think we should use write the content into MemoryStream first. If you just want to be able to delete file upon any exceptions, you can try to create the file first to get the file handler.